motionplan/constraint.go

```diff
@@ -200,118 +303,70 @@ func newCollisionConstraint(
 	}
 
 	// create constraint from reference collision graph
-	constraint := func(cInput *ConstraintInput) (bool, float64) {
-		internal, err := cInput.Frame.Geometries(cInput.StartInput)
+	constraint := func(cInput *StateInput) bool {
+		internal, err := cInput.Frame.Geometries(cInput.Configuration)
 		if err != nil && internal == nil {
-			return false, 0
+			return false
 		}
 
-		cg, err := newCollisionGraph(internal.Geometries(), obstacles, zeroCG, reportDistances)
+		cg, err := newCollisionGraph(internal.Geometries(), obstacles, zeroCG, false)
 		if err != nil {
-			return false, 0
+			return false
 		}
 
 		collisions := cg.collisions()
-		if len(collisions) > 0 {
-			return false, 0
-		}
-		if !reportDistances {
-			return true, 0
-		}
-		sum := 0.
-		for _, collision := range collisions {
-			sum += collision.penetrationDepth
-		}
-		return true, sum
+		return len(collisions) == 0
 	}
 	return constraint, nil
 }
 
 // NewAbsoluteLinearInterpolatingConstraint provides a Constraint whose valid manifold allows a specified amount of deviation from the
 // shortest straight-line path between the start and the goal. linTol is the allowed linear deviation in mm, orientTol is the allowed
 // orientation deviation measured by norm of the R3AA orientation difference to the slerp path between start/goal orientations.
-func NewAbsoluteLinearInterpolatingConstraint(from, to spatial.Pose, linTol, orientTol float64) (Constraint, Metric) {
+func NewAbsoluteLinearInterpolatingConstraint(from, to spatial.Pose, linTol, orientTol float64) (StateConstraint, StateMetric) {
```

Comment:
dumb question why does this return a constraint and metric again?

Reply:
The metric is how cbirrt is able to extend its RRT tree along the valid manifold